src/argilla/client/feedback/dataset/base.py

```diff
@@ -298,19 +178,64 @@ def question_by_name(self, name: str) -> AllowedQuestionTypes:
             f" {', '.join(q.name for q in self._questions)}"
         )
 
-    @property
-    def records(self) -> List[FeedbackRecord]:
-        """Returns the all the records in the dataset."""
-        return self._records + self._new_records
+    def _parse_records(
+        self, records: Union[FeedbackRecord, Dict[str, Any], List[Union[FeedbackRecord, Dict[str, Any]]]]
+    ) -> List[FeedbackRecord]:
+        """Parses the records into a list of `FeedbackRecord` objects.
+
+        Args:
+            records: either a single `FeedbackRecord` or `dict` or a list of `FeedbackRecord` or `dict`.
+
+        Returns:
+            A list of `FeedbackRecord` objects.
+
+        Raises:
+            ValueError: if `records` is not a `FeedbackRecord` or `dict` or a list of `FeedbackRecord` or `dict`.
+        """
+        if isinstance(records, list):
+            if len(records) == 0:
+                raise ValueError("Expected `records` to be a non-empty list of `dict` or `FeedbackRecord`.")
+            new_records = []
+            for record in records:
+                if isinstance(record, dict):
+                    new_records.append(FeedbackRecord(**record))
+                elif isinstance(record, FeedbackRecord):
+                    new_records.append(record)
+                else:
+                    raise ValueError(
+                        "Expected `records` to be a list of `dict` or `FeedbackRecord`,"
+                        f" got type `{type(record)}` instead."
+                    )
+            records = new_records
+        elif isinstance(records, dict):
+            records = [FeedbackRecord(**records)]
+        elif isinstance(records, FeedbackRecord):
+            records = [records]
+        else:
+            raise ValueError(
+                "Expected `records` to be a `dict` or `FeedbackRecord`, got type" f" `{type(records)}` instead."
+            )
+        return records
```

Comment:
I think we can simplify this method, and avoid parsing logic duplication.
```suggestion
        if isinstance(records, (dict, FeedbackRecord)):
            records = [records]

        if len(records) == 0:
            raise ValueError("Expected `records` to be a non-empty list of `dict` or `FeedbackRecord`.")
            
        new_records = []
        for record in records:
            if isinstance(record, dict):
                new_records.append(FeedbackRecord(**record))
            elif isinstance(record, FeedbackRecord):
                new_records.append(record)
            else:
                raise ValueError(
                    "Expected `record` to be a list of `dict` or `FeedbackRecord`,"
                    f" got type `{type(record)}` instead."
                )
        return new_records
```